BaseTools: Remove no longer needed warning suppression

mikebeaton opened the pull request:
# Description

Since https://github.com/tianocore/edk2/commit/ae83c6b7fd83a5906e016a32027c1bcd792a624e `-Wno-unused-but-set-variable`, `-Wno-unused-const-variable` and `-Wno-unused-variable` warning suppression is no longer needed in any builds, and the warnings can be re-enabled to catch real errors.

On pushing this to CI we find two such minors errors introduced in https://github.com/tianocore/edk2/commit/1634dd5b9396702556aa784cfd47c3408d72006f and https://github.com/tianocore/edk2/commit/9c651ef83a74d83219075316aa5d1c189ed608c8, so we fix these in additional commits.

- [ ] Breaking change?
  - NO.
- [ ] Impacts security?
  - NO.
- [ ] Includes tests?
  - NO.

## How This Was Tested

EDK 2 CI (GCC only)
Acidanthera [AUDK CI](https://github.com/acidanthera/audk/commit/2950fc4d6a6c47608a69893b3534a5306524110c) (also tests CLANGDWARF and CLANGPDB)

## Integration Instructions

N/A

mikebeaton commented on BaseTools/Conf/tools_def.template:
```diff
@@ -983,7 +983,7 @@ DEFINE GCC5_LOONGARCH64_PP_FLAGS           = -mabi=lp64d -march=loongarch64 DEF(
 
   DEBUG_GCC48_IA32_CC_FLAGS       = DEF(GCC48_IA32_CC_FLAGS)
-RELEASE_GCC48_IA32_CC_FLAGS       = DEF(GCC48_IA32_CC_FLAGS) -Wno-unused-but-set-variable
+RELEASE_GCC48_IA32_CC_FLAGS       = DEF(GCC48_IA32_CC_FLAGS)
```
No, will test.

ardbiesheuvel commented:
Again, thanks for doing the due diligence.

A while ago, we cloned GCC49 to GCCNOLTO and GCC5 to GCC, in order to decouple our GCC support from the major/minor GCC versions.

Perhaps by the time the upcoming stable tag lands, it's time to
a) get rid of GCC48 and GCC49
b) decide which version of GCC we still want to support, as I suspect that there is little point to supporting anything older than 8 at this time
c) get rid of GCC5, or keep it out of nostalgia, but document that only GCC 8 and older are still supported.

mikebeaton commented:
After discussion and due diligence as above, is this one agreed to merge as is now?

mikebeaton commented:
Rebased, and fixed very minor conflict in two places where changed lines here were right next to lines added for LOONGARCH64 in c33aa63b7dcf7f7377f92becef1a792fe2abecd2. Rebase has also dropped one of the two previously required preceding commits, which has now been merged as 6e5835d76ae27859b761003b6d57ab4ec190428f.

ardbiesheuvel commented on SecurityPkg/Library/HashLibTpm2/HashLibTpm2PeilessSecLib.c:
```diff
@@ -585,7 +584,6 @@ HasLibTpm2PeilessSecLibConstructor (
 
-  TcgPcrEvent             = (TCG_PCR_EVENT *)EventLog;
   TcgEfiSpecIdEventStruct = (TCG_EfiSpecIDEventStruct *)
                             (EventLog + OFFSET_OF (TCG_PCR_EVENT, Event));
```
This should be

```
TcgEfiSpecIdEventStruct = (TCG_EfiSpecIDEventStruct *)TcgPcrEvent->Event;
```

which also fixes the unused warning.

ardbiesheuvel commented on SecurityPkg/Library/HashLibTpm2/HashLibTpm2PeilessSecLib.c:
```diff
@@ -586,9 +586,7 @@ HasLibTpm2PeilessSecLibConstructor (
-  TcgEfiSpecIdEventStruct = (TCG_EfiSpecIDEventStruct *)
-                            (EventLog + OFFSET_OF (TCG_PCR_EVENT, Event));
-
+  TcgEfiSpecIdEventStruct = (TCG_EfiSpecIDEventStruct *)TcgPcrEvent->Event;
```
Yeah better, thanks.

But please drop the URL from the commit log, as it will go stale very quickly, and no need for the stream-of-consciousness.

mikebeaton commented:
Can I cc @mdkinney @os-d @makubacki - Would this change be good to include in the current round of tooling updates? (I think so.)

**Notes**:
 - I have tested that this doesn't break GCC48 (with a more recent, but still somewhat old gcc). That said, I could remove the changes for GCC48, GCC49 and just leave for everything else (GCC, GCC5, XCODE5 (tested), CLANGPDB, CLANGDWARF) if preferred?
 - Also now tested that it doesn't break a fairly large representative sample of edk2-platforms projects, in a sample of toolchains. There's a mountain of stuff there to test completely, but it is certainly not breaking everything left right and centre, or anywhere close.

bexcran commented:
Oh! Yes I did sign up to remove VS2015. I'll try and get back to that soon.

leiflindholm commented:
@vishalo that feels like excessive caution for a trivially locally-revertable change.
